File: src/main/java/com/pandy/base/array/Compute_PI.py
import math
import random


def simulator(n):
    n_in_a_circle = 0
    for i in range(n):
        # random.().random(): (0, 1)
        x = random.random()
        y = random.random()
        if math.sqrt(x*x + y*y) < 1:
            n_in_a_circle += 1
    return n_in_a_circle


def compute_pi(n_in_a_circle, n_in_a_square):
    return 4 * n_in_a_circle / n_in_a_square

if __name__ == '__main__':
    n = 7000000
    m = simulator(n)
    pi = 4 * m / n
    print(pi)

File: src/main/java/com/pandy/base/array/Compute_PI.java
package com.pandy.base.array;


public class Compute_PI {
    public static void main(String[] args) {
        Compute_PI test = new Compute_PI();

//        int i = test.sumSubString("a ", " ");
        double i =  test.calculate_it();

        System.out.println("i = " + i);
        System.out.println("i2 = " + 3.14/4);
    }

    public double calculate_it()
    {
        double x = 0.0d;
        double y = 0.0d;
        int total = 0;
        for(int i = 0;i<7000000;i++){
            x = Math.random();
            y = Math.random();
            if(Math.sqrt(x*x+y*y) < 1)
                total++;
        }
        return total/7000000.0;
    }
}
